pandas/core/generic.py

```diff
@@ -637,12 +637,17 @@ def _get_cleaned_column_resolvers(self) -> dict[Hashable, Series]:
         Used in :meth:`DataFrame.eval`.
         """
         from pandas.core.computation.parsing import clean_column_name
+        from pandas.core.series import Series
 
         if isinstance(self, ABCSeries):
             return {clean_column_name(self.name): self}
 
         return {
-            clean_column_name(k): v for k, v in self.items() if not isinstance(k, int)
+            clean_column_name(k): Series(
+                v._values, copy=False, index=v.index
+            ).__finalize__(v)
+            for k, v in self.items()
```

Comment:
Maybe you can use `_iter_column_arrays` here instead of `items()` to avoid going through Series construction twice?

Reply:
Yep that should work, thx

Reply:
With my suggestion, the column name was lost. Hopefully the last commit should fix that